osquery/tables/system/windows/services.cpp

```diff
@@ -0,0 +1,167 @@
+                                  "PAUSE_PENDING",
+                                  "PAUSED"};
+
+const std::map<int, std::string> kServiceType = {
+    {0x00000010, "OWN_PROCESS"},
+    {0x00000020, "SHARE_PROCESS"},
+    {0x00000100, "INTERACTIVE_PROCESS"},
+    {0x00000110, "OWN_PROCESS   (Interactive)"},
+    {0x00000120, "SHARE_PROCESS (Interactive)"}};
+
+BOOL QuerySvcInfo(const SC_HANDLE& schSCManager,
+                  ENUM_SERVICE_STATUS_PROCESS& svc,
+                  Row& r) {
+  SC_HANDLE schService;
+  LPQUERY_SERVICE_CONFIG lpsc = nullptr;
+  LPSERVICE_DESCRIPTION lpsd = nullptr;
+  DWORD cbBufSize = 0;
+
+  schService =
+      OpenService(schSCManager, svc.lpServiceName, SERVICE_QUERY_CONFIG);
+
+  if (schService == nullptr) {
+    TLOG << "OpenService failed (" << GetLastError() << ")";
+    return FALSE;
+  }
+
+  (void)QueryServiceConfig(schService, nullptr, 0, &cbBufSize);
+  lpsc = (LPQUERY_SERVICE_CONFIG)malloc(cbBufSize);
+  if (!QueryServiceConfig(schService, lpsc, cbBufSize, &cbBufSize)) {
+    TLOG << "QueryServiceConfig failed (" << GetLastError() << ")";
+  }
+
+  (void)QueryServiceConfig2(
+      schService, SERVICE_CONFIG_DESCRIPTION, nullptr, 0, &cbBufSize);
+  lpsd = (LPSERVICE_DESCRIPTION)malloc(cbBufSize);
+  if (!QueryServiceConfig2(schService,
+                           SERVICE_CONFIG_DESCRIPTION,
+                           (LPBYTE)lpsd,
+                           cbBufSize,
+                           &cbBufSize)) {
+    TLOG << "QueryServiceConfig2 failed (" << GetLastError() << ")";
+  }
+
+  r["name"] = SQL_TEXT(svc.lpServiceName);
+  r["display_name"] = SQL_TEXT(svc.lpDisplayName);
+  r["status"] = SQL_TEXT(kSvcStatus[svc.ServiceStatusProcess.dwCurrentState]);
+  r["pid"] = INTEGER(svc.ServiceStatusProcess.dwProcessId);
+  r["win32_exit_code"] = INTEGER(svc.ServiceStatusProcess.dwWin32ExitCode);
+  r["service_exit_code"] =
+      INTEGER(svc.ServiceStatusProcess.dwServiceSpecificExitCode);
+  r["start_type"] = SQL_TEXT(kSvcStartType[lpsc->dwStartType]);
+  r["path"] = SQL_TEXT(lpsc->lpBinaryPathName);
+  r["user_account"] = SQL_TEXT(lpsc->lpServiceStartName);
+
+  if (lpsd->lpDescription != nullptr) {
+    r["description"] = SQL_TEXT(lpsd->lpDescription);
+  }
+
+  if (kServiceType.count(lpsc->dwServiceType) > 0) {
+    r["service_type"] = SQL_TEXT(kServiceType.at(lpsc->dwServiceType));
+  } else {
+    r["service_type"] = SQL_TEXT("UNKNOWN");
+  }
+
+  QueryData regResults;
+  queryKey("HKEY_LOCAL_MACHINE",
+           "SYSTEM\\CurrentControlSet\\Services\\" + r["name"] + "\\Parameters",
+           regResults);
+  for (const auto& aKey : regResults) {
+    if (aKey.at("name") == "ServiceDll") {
+      r["module_path"] = SQL_TEXT(aKey.at("data"));
+    }
+  }
+
+  free(lpsc);
+  free(lpsd);
+  CloseServiceHandle(schService);
+  return TRUE;
+}
+
+QueryData genServices(QueryContext& context) {
+  SC_HANDLE schSCManager;
+  void* buf = nullptr;
+  DWORD BytesNeeded = 0;
```

Comment:
Nit, typically for local variables, you'll want a camelCase naming scheme to avoid confusion with classes.